### PR TITLE
NIAD-382: Example Transactions

### PR DESCRIPTION
**Pull request by mattd-kainos**
* Moved instructions about available examples into the README
* Moved inbound examples into UAT data folder
* UAT skips notes and 'ignore' files
* Modified inbound/rejection/2 to reflect the actual permitted character set

### CODE REVIEW

**Review by bartek-sarul — DISMISSED** [review]

**Review by bartek-sarul — APPROVED** [review]

File: src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java [hunk 1]
```diff
@@ -33,6 +33,8 @@ public CustomArgumentsProvider(String folder) {
         var resources = getResources();
 
         var grouped = Arrays.stream(resources)
+            .filter(r -> !r.getFilename().endsWith("txt")) // ignore notes
```
Comment (michael-kainos) [inline src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java:36]: it's not clear what the purpose of the parameter r is

File: src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java [hunk 2]
```diff
@@ -43,7 +45,7 @@ public CustomArgumentsProvider(String folder) {
             .peek(es -> {
```
Comment (michael-kainos) [inline src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java:45]: same here, not sure on the purpose of es

File: src/intTest/resources/inbound_uat_data/rejection/2.edifact.ignore.dat [hunk 3]
```diff
@@ -7,6 +7,6 @@ RFF+950:F3'
 S01+1'
 RFF+TN:12'
 NAD+GP+4826940,281:900'
-FTX+RGI+++WRONG HA - TRY SURREY'
+FTX+RGI+++ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890 .,-()/=!"%&*;<>?'?+?:??'
```
Comment (michael-kainos) [inline src/intTest/resources/inbound_uat_data/rejection/2.edifact.ignore.dat:10]: + character is not valid here

File: src/intTest/resources/inbound_uat_data/rejection/2.fhir.ignore.json [hunk 4]
```diff
@@ -27,7 +23,11 @@
     },
     {
       "name": "freeText",
-      "valueString": "SPECIAL CHARACTERS ALLOWED EXCEPT PLUS SIGN !@£$%^&*()?><:\"|\\#][{}-_="
+      "valueString": "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890 .,-()/=!\"%&*;<>?'?+?:??"
```
Comment (michael-kainos) [inline src/intTest/resources/inbound_uat_data/rejection/2.fhir.ignore.json:26]: same here, i don't think + is allowed?

File: src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java (hunk 1 above)
Comment (mattd-kainos) [reply src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java:36]: ignores files that end in txt - those are notes as described in the readme

File: src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java (hunk 2 above)
Comment (mattd-kainos) [reply src/intTest/java/uk/nhs/digital/nhsconnect/nhais/uat/CustomArgumentsProvider.java:45]: abbreviations of parameter names to lambdas / streams are usually considered ok

**Review by michael-kainos — APPROVED** [review]

**Pull request merged**